server/src/main/java/org/jboss/as/server/services/net/LocalDestinationOutboundSocketBindingAddHandler.java

```diff
@@ -71,6 +79,30 @@ protected void performRuntime(final OperationContext context, final ModelNode op
 
         final PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
         final String outboundSocketName = address.getLastElement().getValue();
+
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                final String currentSocketBindingGroupName = address.getParent().getLastElement().getValue();
+                final Set<ResourceEntry> children = context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS)
+                        .getChildren(ModelDescriptionConstants.SOCKET_BINDING_GROUP);
+                for (Resource.ResourceEntry re : children) {
+                    if (re.getName().equalsIgnoreCase(currentSocketBindingGroupName)) {
+                        for (Resource.ResourceEntry entry : re.getChildren(ModelDescriptionConstants.SOCKET_BINDING)) {
+                            String name = entry.getName();
+                            if (name.equalsIgnoreCase(outboundSocketName))
+                                throw ControllerLogger.ROOT_LOGGER.socketBindingalreadyDeclared(
+                                        Element.SOCKET_BINDING.getLocalName(), Element.OUTBOUND_SOCKET_BINDING.getLocalName(),
+                                        name, Element.SOCKET_BINDING_GROUP.getLocalName(), currentSocketBindingGroupName);
+
+                        }
+                        break;
+                    }
+                }
+            }
+
+        }, Stage.RUNTIME);
```

Comment:
This is model validation and hence needs to be in Stage.MODEL. So it needs to be moved out of performRuntime().